codec/reflectcodec/type_codec.go

```diff
@@ -235,6 +239,10 @@ func (c *genericCodec) size(
 			return 0, false, err
 		}
 
+		if keySize == 0 && valueSize == 0 {
```

Comment:
Do we not need to check this for remaining keys/value pairs?

Reply:
If the size is 0 of an instance, it is guaranteed to always be 0 based on the type (because we don't have optional fields).